### PR TITLE
Implemented EhCaching for our historicalPrices endpoint (with logging)

### PR DESCRIPTION
**Pull request by RamiDaoud**

### CODE REVIEW

File: src/main/java/org/galatea/starter/service/IexService.java [hunk 1]
```diff
@@ -89,10 +92,12 @@ private String getToken() {
    * @return a List of IexHistoricalPrices objects for the given symbol,
    *         from the given date until the given date plus range
    */
+  @Cacheable(cacheNames = "historical", sync = true)
   public List<IexHistoricalPrices> getHistoricalPrices(
       final String symbol,
       final String range,
       final String date) {
+    log.info("DID NOT HIT CACHE");
```
Comment (jbrody17) [inline src/main/java/org/galatea/starter/service/IexService.java]: Nice log. Should probably clean up a bit (not all caps)

File: src/main/java/org/galatea/starter/service/IexService.java [hunk 2]
```diff
@@ -59,6 +60,7 @@ private String getToken() {
    *
    * @return a list of all Stock Symbols from IEX.
    */
+  @Cacheable(cacheNames = "historical", sync = true)
```
Comment (jbrody17) [inline src/main/java/org/galatea/starter/service/IexService.java]: Why are you setting sync = true?

**Review by jbrody17 — COMMENTED** [review]
These cacheable methods will work given an exact request. What if we had a subset of symbols cached

File: src/main/java/org/galatea/starter/service/IexService.java (hunk 1 above)
Comment (RamiDaoud) [reply src/main/java/org/galatea/starter/service/IexService.java]: Sure!

**Pull request merged**